tt_metal/impl/dispatch/kernels/cq_prefetch.cpp

```diff
@@ -1124,10 +1169,19 @@ bool process_cmd(uint32_t& cmd_ptr,
 
     case CQ_PREFETCH_CMD_RELAY_INLINE:
         //DPRINT << "relay inline" << ENDL();
-        if (exec_buf) {
-            stride = process_exec_buf_relay_inline_cmd(cmd_ptr, downstream_data_ptr, exec_buf_state);
+        if constexpr (exec_buf) {
+            if (cmd->relay_inline.dispatcher_type == DispatcherSelect::DISPATCH_MASTER) {
+                stride = process_exec_buf_relay_inline_cmd<true>(cmd_ptr, downstream_data_ptr, exec_buf_state);
```

Comment:
right here (for example) pass the last 2 args in as template args instead of the bool

Reply:
Addressed above.